src/core/mac/sub_mac.cpp

```diff
@@ -1160,33 +1123,22 @@ void SubMac::HandleCslTimer(void)
         else
         {
             mCslTimer.FireAt(mCslSampleTime + timeAfter);
-            mCslState = kCslSample;
+            mIsCslSampling = true;
             mCslSampleTime += periodUs;
         }
 
         Get<Radio>().UpdateCslSampleTime(mCslSampleTime.GetValue());
 
         if (RadioSupportsReceiveTiming())
         {
-            if (mState != kStateDisabled && mCslChannel)
-            {
-                IgnoreError(Get<Radio>().ReceiveAt(mCslChannel, mCslSampleTime.GetValue() - periodUs - timeAhead,
-                                                   timeAhead + timeAfter));
-            }
+            IgnoreError(Get<Radio>().ReceiveAt(mCslChannel, mCslSampleTime.GetValue() - periodUs - timeAhead,
```

Comment:
I noticed you removed the `mState != kStateDisabled` check.

I was checking if we can get here when we are in `kStateDisabled`. 
- `mCslTimer` can be started when `UpdateCsl()` is called (and we have non-zero period).
-  From `SubMac::Disable()` we don't stop the CSL timer.
-  So I guess we can technically get here in any `mState`.  
- May be safer if we still keep the `mState != kStateDisabled` check (don't want to call to `Radio` if it is disabled)

I think we don't The `mCslChannel` check since `UpdateCsl()` will always set the channel to something.

Reply:
Yes, it will be safer. In any case, what do you thing about stopping the timer in `SubMac::Disable()`?

Reply:
Yes. Good idea to stop the timer in `Disable()`.

But I would suggest keeping the `mState != kStateDisabled` check here to be safe.
(e.g. maybe `UpdateCsl()` is called (user sets a new CSL period) and restart the timer while we are in disabled state).

Reply:
Done.